content-src/components/DiscoveryStreamComponents/DSCard/DSCard.jsx

```diff
@@ -1,12 +1,69 @@
-import {actionCreators as ac} from "common/Actions.jsm";
+import {actionCreators as ac, actionTypes as at} from "common/Actions.jsm";
 import React from "react";
 
+const VISIBLE = "visible";
+const VISIBILITY_CHANGE_EVENT = "visibilitychange";
+const INTERSECTION_RATIO = 0.5;
+
 export class DSCard extends React.PureComponent {
   constructor(props) {
     super(props);
+
+    this.setupIntersectionObserver = this.setupIntersectionObserver.bind(this);
+    this.onIntersectionObserve = this.onIntersectionObserve.bind(this);
+    this.isIntersecting = this.isIntersecting.bind(this);
+    this.cardElementRef = this.cardElementRef.bind(this);
     this.onLinkClick = this.onLinkClick.bind(this);
   }
 
+  componentDidMount() {
+    if (this.props.document.visibilityState === VISIBLE) {
+      this.setupIntersectionObserver();
+    } else {
+      this._onVisibilityChange = () => {
+        if (this.props.document.visibilityState === VISIBLE) {
+          this.setupIntersectionObserver();
```

Comment:
nit: `this.setupIntersectionObserver.bind(this)` is not needed here since the fat arrow function automatically binds this for you.

Reply:
I'm not longer using the arrow for this anymore.

Reply:
`this._onVisibilityChange = () => {...`, and you bound `this` here https://github.com/mozilla/activity-stream/pull/4731/files#diff-cb93e859d1f25eeace50eb38edb98cd7R12